benchmark/single-source/BufferFind.swift

```diff
@@ -0,0 +1,108 @@
+//===--- BufferFind.swift -------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2021 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import TestsUtils
+
+public var benchmarks: [BenchmarkInfo] = [
+  // size 1000, alignment 0
+  BenchmarkInfo(
+    name: "RawBuffer.1000.findFirst",
+    runFunction: run_BufferFindFirst,
+    tags: [.validation, .api],
+    setUpFunction: buffer1000Setup,
+    tearDownFunction: bufferTeardown
+  ),
+  BenchmarkInfo(
+    name: "RawBuffer.1000.findLast",
+    runFunction: run_BufferFindLast,
+    tags: [.validation, .api],
+    setUpFunction: buffer1000Setup,
+    tearDownFunction: bufferTeardown
+  ),
+  // size 15, alignment 0
+  BenchmarkInfo(
+    name: "RawBuffer.15.findFirst",
+    runFunction: run_BufferFindFirst,
+    tags: [.validation, .api],
+    setUpFunction: buffer15Setup,
+    tearDownFunction: bufferTeardown
+  ),
+  BenchmarkInfo(
+    name: "RawBuffer.15.findLast",
+    runFunction: run_BufferFindLast,
+    tags: [.validation, .api],
+    setUpFunction: buffer15Setup,
+    tearDownFunction: bufferTeardown
+  ),
+  // size 7, alignment 0
+  BenchmarkInfo(
+    name: "RawBuffer.7.findFirst",
+    runFunction: run_BufferFindFirst,
+    tags: [.validation, .api],
+    setUpFunction: buffer7Setup,
+    tearDownFunction: bufferTeardown
+  ),
+  BenchmarkInfo(
+    name: "RawBuffer.7.findLast",
+    runFunction: run_BufferFindLast,
+    tags: [.validation, .api],
+    setUpFunction: buffer7Setup,
+    tearDownFunction: bufferTeardown
+  )
+]
+
+var buffer: UnsafeMutableRawBufferPointer = .init(start: nil, count: 0)
+
+func buffer1000Setup() {
+  bufferSetup(size: 1000, alignment: 0)
+}
+
+func buffer15Setup() {
+  bufferSetup(size: 15, alignment: 0)
+}
+
+func buffer7Setup() {
+  bufferSetup(size: 7, alignment: 0)
+}
+
+func bufferTeardown() {
+  buffer.deallocate()
+  buffer = .init(start: nil, count: 0)
+}
+
+func bufferSetup(size: Int, alignment: Int) {
+  buffer = UnsafeMutableRawBufferPointer.allocate(byteCount: size, alignment: alignment)
+  buffer.initializeMemory(as: UInt8.self, repeating: UInt8.min)
+  buffer[size / 2] = UInt8.max
+}
+
+@inline(never)
+public func run_BufferFindFirst(n: Int) {
+  var offset = 0
+  for _ in 0 ..< n * 1000_000 {
```

Comment:
Let's make this multiplier 10_000

Reply:
done

Reply:
Here was the output from the benchmark runner with the 1_000_000 multiplier:


| ✅    | Benchmark Check Report                                       |
| ---- | ------------------------------------------------------------ |
| ⚠️⏱   | `RawBuffer.7.findFirst` execution took at least 4335 μs. <br><sub> Decrease the workload of `RawBuffer.7.findFirst` by a factor of 8 (10), to be less than 1000 μs. <br><sub> unable to compute memory footprint of `RawBuffer.7.findFirst` |
| ⛔️⏱   | `RawBuffer.1000.findLast` execution took at least 282237 μs. <br><sub> Decrease the workload of `RawBuffer.1000.findLast` by a factor of 512 (1000), to be less than 1000 μs. <br><sub> unable to compute memory footprint of `RawBuffer.1000.findLast` |
| ⚠️⏱   | `RawBuffer.15.findLast` execution took at least 5665 μs. <br><sub> Decrease the workload of `RawBuffer.15.findLast` by a factor of 8 (10), to be less than 1000 μs. <br><sub> unable to compute memory footprint of `RawBuffer.15.findLast` |
| ⚠️⏱   | `RawBuffer.7.findLast` execution took at least 3486 μs. <br><sub> Decrease the workload of `RawBuffer.7.findLast` by a factor of 4 (10), to be less than 1000 μs. <br><sub> unable to compute memory footprint of `RawBuffer.7.findLast` |
| ⚠️⏱   | `RawBuffer.15.findFirst` execution took at least 7626 μs. <br><sub> Decrease the workload of `RawBuffer.15.findFirst` by a factor of 8 (10), to be less than 1000 μs. <br><sub> unable to compute memory footprint of `RawBuffer.15.findFirst` |
| ⛔️⏱   | `RawBuffer.1000.findFirst` execution took at least 448424 μs. <br><sub> Decrease the workload of `RawBuffer.1000.findFirst` by a factor of 512 (1000), to be less than 1000 μs. <br><sub> unable to compute memory footprint of `RawBuffer.1000.findFirst` |